src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopTerasortExecutor.cs

```diff
@@ -0,0 +1,389 @@
+                                this.ExecuteCommandAsync("bash", $"-c sbin/stop-yarn.sh", this.PackageDirectory, telemetryContext, CancellationToken.None),
+                                this.stateManager.SaveStateAsync<HadoopExecutorState>($"{nameof(HadoopExecutorState)}", this.state, CancellationToken.None)).ConfigureAwait(false);
+                        }
+                    });
+                }).GetAwaiter().GetResult(); // Ensure Task.Run completes before continuing
+
+                this.disposed = true;
+            }
+        }
+
+        /// <summary>
+        /// Executes the Hadoop Terasort workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string timestamp = DateTime.Now.ToString("ddMMyyHHmmss");
+            string teragenCommmand = $"bin/hadoop jar share/hadoop/mapreduce/hadoop-mapreduce-examples-3.3.5.jar teragen {this.RowCount} /inp-{timestamp}-{this.ProfileIteration}";
+            string terasortCommand = $"bin/hadoop jar share/hadoop/mapreduce/hadoop-mapreduce-examples-3.3.5.jar terasort /inp-{timestamp}-{this.ProfileIteration} /out-{timestamp}-{this.ProfileIteration}";
+
+            await this.RunSortingOperationsAsync(teragenCommmand, "Hadoop Teragen", telemetryContext, cancellationToken).ConfigureAwait(false);
+            await this.RunSortingOperationsAsync(terasortCommand, "Hadoop Terasort", telemetryContext, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the Hadoop Terasort workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.state = await this.stateManager.GetStateAsync<HadoopExecutorState>($"{nameof(HadoopExecutorState)}", cancellationToken).ConfigureAwait(false)
+                ?? new HadoopExecutorState();
+
+            DependencyPath workloadPackage = await this.packageManager.GetPlatformSpecificPackageAsync(
+                this.PackageName, this.Platform, this.CpuArchitecture, cancellationToken).ConfigureAwait(false);
+
+            DependencyPath javaPackage = await this.GetPackageAsync(
+                this.JdkPackageName, cancellationToken).ConfigureAwait(false);
+
+            this.PackageDirectory = workloadPackage.Path;
+            this.JavaPackageDirectory = javaPackage.Path;
+
+            this.JavaExecutablePath = this.PlatformSpecifics.Combine(this.JavaPackageDirectory, "bin", "java");
+            this.ExecutablePath = this.PlatformSpecifics.Combine(this.PackageDirectory, "bin", "hadoop");
+
+            if (!this.state.HadoopExecutorStateInitialized)
+            {
+                this.SetEnvironmentVariable(EnvironmentVariable.JAVA_HOME, this.JavaPackageDirectory, EnvironmentVariableTarget.Process);
```

Comment:
This should already be done at the jdk package installation

Reply:
Yes, but the values were not updated in the expected manner during the testing. Based on the purpose of the workload, I have added it here.